packages/changed-elements-react/src/widgets/comparisonJobWidget/hooks/useNamedVersionLoader.tsx

```diff
@@ -175,176 +158,59 @@ const sortAndSetIndexOfNamedVersions = (namedVersions: NamedVersion[], currentNa
     onError();
     return;
   }
-  const reversedNamedVersions = namedVersionsOlderThanCurrentVersion.reverse()
+  const reversedNamedVersions = namedVersionsOlderThanCurrentVersion.reverse();
   if (reversedNamedVersions[0].changesetIndex === currentNamedVersion.changesetIndex) {
     reversedNamedVersions.shift(); //remove current named version
   }
-  const changeSetMap = new Map<number, Changeset>();
-  changesets.forEach((changeset: Changeset) => {
-    changeSetMap.set(changeset.index, changeset);
-  })
+  const changeSetMap = arrayToMap(changesets, (changeSet: Changeset) => { return changeSet.index; });
   // we must offset the named versions , because that changeset is "already applied" to the named version, see this:
   // https://developer.bentley.com/tutorials/changed-elements-api/#221-using-the-api-to-get-changed-elements
   // this assuming latest is current
   const offSetNameVersions = reversedNamedVersions.map((version) => {
     version.changesetIndex = version.changesetIndex + 1;
-    version.changesetId = changeSetMap.get(version.changesetIndex)?.id ?? version.changesetId
+    version.changesetId = changeSetMap.get(version.changesetIndex)?.id ?? version.changesetId;
     return version;
-  })
+  });
   return offSetNameVersions;
 };
 
 type ProcessChangesetsArgs = {
   iTwinId: string;
   iModelId: string;
-  namedVersionLoaderState: NamedVersionLoaderState;
+  namedVersionLoaderState: NamedVersionLoaderResult;
```

Comment:
You replaced the type used here, did you miss also renaming the property name? (You might not have, which is fine.)